frontend/src/api/auth.ts

```diff
@@ -1,13 +1,34 @@
-import axios from "axios";
+import axios, { AxiosError } from "axios";
 import { User } from "../model/User";
 
 export const login = async (username: string, password: string) => {
     try {
-        const response = await axios.post("/api/auth/login", {username,password});
+        const response = await axios.post("/api/auth/login", { username, password });
         if (response.status === 200) {
             return response.data as User;
         }
     } catch (err) {
         return null;
     }
-};
\ No newline at end of file
+};
+
+export const register = async (username: string, email: string, password: string, confirm: string) => {
+    if(password !== confirm) {
+        return { status: 400, message: "Salasanat eivät täsmää"};
+    }
+    try {
+        const response = await axios.post("/api/auth/register", { username, password, email });
+        if (response.status == 200) {
+            const user: User = response.data as User;
+            return { status: 200, data: user };
+        }
+        else {
+            return { status: 400, message: "Ei sallittu" };
```

Comment:
Possibly line 26 could be: return { status: response.status, message: "Ei sallittu" }; but basically we should never come to this line, because when axios call is successful, we already returned in line 23 and when it is not successful, we return from catch(err) segment in line 32. Actually line 26 was needed, because when it was not there, React didn't accept the code, because returned value could be 'undefined' in that case, when status is not 200, but there is no error. (Although we know that we should actually never come to the line 26).